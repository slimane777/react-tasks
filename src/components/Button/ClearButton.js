import React from 'react'
import { Button, Wrapper } from './styles'
import TasksCount from './TasksCount';
import { useSelector, useDispatch } from 'react-redux';
import {
  DELETE_TASK_REQUESTED
} from '../../redux/actions/tasks'

const ClearButton = () => {

  const count = useSelector((state) => state.tasks.tasks)
  const dispatch = useDispatch()

  const deleteTask = () => dispatch({ type: DELETE_TASK_REQUESTED, payload: 5 });

  return (
    <Wrapper>
      <TasksCount count={count.length} />
      <Button className={'clear'} onClick={() => deleteTask()}>
          Clear List
      </Button>
    </Wrapper>
  )
}

export default ClearButton;