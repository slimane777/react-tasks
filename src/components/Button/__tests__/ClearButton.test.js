import React from 'react';
import renderer from 'react-test-renderer';
import ClearButton from '../ClearButton';
import Enzyme, { shallow } from 'enzyme';
import sinon from 'sinon';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('ClearButton test', () => {
  const initialState = {tasks:{tasks: []}}
  const mockStore = configureStore()
  let store,wrapper

  it('ClearButton should match snapshot', () => {
    store = mockStore(initialState)
    const component = renderer.create(<Provider store={store}><ClearButton
      /></Provider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('simulates click events', () => {
  //   const mockCallBack = sinon.spy();
  //   const button = shallow(<Provider store={store} ><ClearButton onClick={mockCallBack}
  //     /></Provider>);

  //     button.find('.clear').simulate('click');
  //   expect(mockCallBack).toHaveProperty('callCount', 1);
  // });

});
