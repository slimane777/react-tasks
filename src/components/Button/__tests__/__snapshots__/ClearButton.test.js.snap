// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClearButton test ClearButton should match snapshot 1`] = `
<div
  className="sc-dlnjwi cvjzis"
>
  <span
    className="sc-gtsrHT jOoloQ"
  >
    0
     Tasks
  </span>
  <p
    className="sc-bdnxRM ckcDZN clear"
    onClick={[Function]}
  >
    Clear List
  </p>
</div>
`;
